cranelift/codegen/src/inst_predicates.rs

```diff
@@ -41,9 +41,11 @@ pub fn has_side_effect(func: &Function, inst: Inst) -> bool {
     trivially_has_side_effects(opcode) || is_load_with_defined_trapping(opcode, data)
 }
 
-/// Does the given instruction have any side-effect as per [has_side_effect], or else is a load?
-pub fn has_side_effect_or_load(func: &Function, inst: Inst) -> bool {
-    has_side_effect(func, inst) || func.dfg[inst].opcode().can_load()
+/// Does the given instruction have any side-effect as per [has_side_effect], or else is a load,
+/// but not the get_pinned_reg opcode?
+pub fn has_side_effect_or_load_not_get_pinned_reg(func: &Function, inst: Inst) -> bool {
```

Comment:
I'm ok to do this, but since the helper is unused in other contexts that don't want to check for get_pinned_reg, this would be over-generalizing the helper without any clear benefits. Should we instead move this helper to the `machinst` directory, and rename it `has_lowering_side_effect` or something like this instead?

Reply:
Hmm, yeah, that sounds fine. I think the thing that stuck out the most to me was the the function name essentially had the same complexity as its body :-) "Lowering side effect" is much better.